soh/soh/Enhancements/mods.cpp

```diff
@@ -1223,6 +1224,22 @@ void RegisterToTMedallions() {
     });
 }
 
+void RegisterFloorSwitchesHook() {
+    GameInteractor::Instance->RegisterGameHook<GameInteractor::OnActorInit>([](void* refActor) {
+        Actor* actor = static_cast<Actor*>(refActor);
+        if (actor->id == ACTOR_OBJ_SWITCH) {
+            ObjSwitch* switchActor = reinterpret_cast<ObjSwitch*>(actor);
+            s32 type = (switchActor->dyna.actor.params & 7);
+
+            if (CVarGetInteger("gEnhancements.FixFloorSwitches", 0)) {
+                if (switchActor->dyna.actor.params == 4608 || switchActor->dyna.actor.params == 14848) {
+                    switchActor->dyna.actor.world.pos.y -= 1;
+                }
+            }
+        }
+    });
+}
+
```

Comment:
Hey! The current setup looks great. I have a small suggestion for the hook section.  I think incorporate an early return to avoid repetitive checks and introduce a switch statement could be a good idea.
This way, it'll be easier to add more elements in the future for improved visibility

```suggestion
void RegisterFloorSwitchesHook() {
    GameInteractor::Instance->RegisterGameHook<GameInteractor::OnActorInit>([](void* refActor) {
        Actor* actor = static_cast<Actor*>(refActor);
        
        if (actor->id == ACTOR_OBJ_SWITCH) {
            ObjSwitch* switchActor = reinterpret_cast<ObjSwitch*>(actor);
            s32 type = switchActor->dyna.actor.params & 7;

            if (!CVarGetInteger("gEnhancements.FixFloorSwitches", 0)) {
                return;
            }

            switch (switchActor->dyna.actor.params) {
                case 4608:
                case 14848:
                    switchActor->dyna.actor.world.pos.y -= 1;
                    break;

                default:
                    break;
            }
        }
    });
}

```